### PR TITLE
Adapt neurochem trainer to NC setup

### LINKED ISSUE #243: Make torchani's training setup identical to neurochem
After a thorough comparison between Neurochem and Torchani, I have found a few differences as explained briefly in the following:

- [x] NC skips the weight decay for the first layer of each atomic network, even if the weight decay parameter is provided in the input. This was the main source of difference in training performance.(#275)
- [x] In NC the weights are initialized to He Kaiming normal and the biases to zero. (#245, #275)
- [x] Loss in NC is scaled by sqrt(natoms). (#275)
- [x] NC uses AdamW optimizer to update the weights and SGD for the biases. (#275)
- [x] There were a couple of discrepancies between theory and the implementation of AdamW in Torchani (#261).
- [x] NC scale the gradients of atomic networks by the number of atoms in that batch, and compute the gradient from SumSquaredError instead of MeanSquaredError. (will be fixed in Neurochem)
- [ ] NC reshuffle the batches after every epoch (tracked in #248, no much improvement as tested, low priority)

### PR DESCRIPTION
**Pull request by farhadrgh**
Here is the summary of the changes in `torchani/neurochem/__init__.py` in order to adapt to the current NC setup and fix #243.

- In NC the weights are initialized to He Kaiming normal and the biases to zero. 
- Loss is scaled by sqrt(natoms).
- AdamW optimizer is used to update weights and SGD for the biases.
- Ignite is ignored.
- Warnings are removed.

### CODE REVIEW

**Comment by farhadrgh** [comment]
> This part LGTM. But do we still need to find a way to scale the gradients, and shuffle the dataset every epoch?

I have scaled the gradients in my scripts, but scaling gradient in NC was something that was not intended. So it is going to be fixed in Neurochem. And I thought we don't need to include it in Torchani.

I have tested dataset shuffling in every epoch, and since that did not introduce performance enhancement, we can skip it for now?

**Comment by zasdfgbnm** [comment]
> I have scaled the gradients in my scripts, but scaling gradient in NC was something that was not intended. So it is going to be fixed in Neurochem. And I thought we don't need to include it in Torchani.

This LGTM

> I have tested dataset shuffling in every epoch, and since that did not introduce performance enhancement, we can skip it for now?

I think we could make this a low priority task. Our dataset API is currently horrible anyway (for example it requires tons of RAM).
